blockstack_client/app.py

```diff
@@ -53,60 +58,64 @@ def app_register( name, app_name, app_account_id, app_url, app_storage_drivers=N
     """
 
     if not interactive and password is None:
-        raise ValueError("Non-interactive use requires a password")
+        raise ValueError('Non-interactive use requires a password')
 
     config_dir = os.path.dirname(config_path)
     app_wallet_dir = os.path.join(config_dir, APP_WALLET_DIRNAME)
-    if not os.path.exists( app_wallet_dir ):
+    if not os.path.exists(app_wallet_dir):
         try:
-            os.makedirs(app_wallet_dir, 0700 )
-        except Exception, e:
+            os.makedirs(app_wallet_dir, 0700)
+        except Exception as e:
             log.exception(e)
-            return {'error': "Failed to create app wallet directory"}
+            return {'error': 'Failed to create app wallet directory'}
 
     # create the wallet
-    wallet_path = app_wallet_path( config_dir, name, app_name, app_account_id )
+    wallet_path = app_wallet_path(config_dir, name, app_name, app_account_id)
     if os.path.exists(wallet_path):
-        return {'error': 'Wallet already exists (%s)' % wallet_path}
+        return {'error': 'Wallet already exists ({})'.format(wallet_path)}
 
     if password is None:
-        assert interactive
-        password = ""
+        prompt = 'Creating new application wallet'
+        password = ''
         while len(password) < config.WALLET_PASSWORD_LENGTH:
-            res = wallet.make_wallet_password( prompt="Creating new application wallet", password=password )
-            if 'error' in res:
-                print res['error']
-                continue
-            else:
+            res = wallet.make_wallet_password(prompt=prompt, password=password)
+            if 'error' not in res:
                 password = res['password']
                 break
-            
+            print(res['error'])
+
     pk_hex = keylib.ECPrivateKey().to_hex()
-    res = wallet.initialize_wallet(password=password, interactive=interactive, hex_privkey=pk_hex, config_dir=config_dir, wallet_path=wallet_path )
+    res = wallet.initialize_wallet(
+        password=password, interactive=interactive,
+        hex_privkey=pk_hex, config_dir=config_dir, wallet_path=wallet_path
+    )
     if 'error' in res:
-        log.error("Failed to create wallet '%s.%s@%s': %s" % (app_account_id, app_name, name, res['error']))
+        msg = 'Failed to create wallet "{}.{}@{}": {}'
+        log.error(msg.format(app_account_id, app_name, name, res['error']))
         return res
 
     pub_hex = keylib.ECPrivateKey(pk_hex).public_key().to_hex()
 
     # preferred storage drivers?
     conf = config.get_config(config_path)
-    if app_storage_drivers is None:
-        app_storage_drivers = conf['storage_drivers']
-
-    if proxy is None:
-        proxy = get_default_proxy(config_path=config_path)
+    app_storage_drivers = app_storage_drivers or conf['storage_drivers']
```

Comment:
>  Is there a chance that this could accidentally evaluate to a boolean, such as by way of misplaced parenthesis?
No, that won't happen. See below.

I've found that treating '', None, [] and {} in this manner works well in practice. Because the subsequent lines throw an error when some variable is used incorrectly.

However, I can use the explicit check for None. If that's what you prefer.

```
>>> False or {'key':'value'}['key']
'value'
>>> (False or {'key':'value'}['key'])
'value'
>>> 
```